src/email/validators.ts

```diff
@@ -4,3 +4,29 @@ export const isEmail = (arg: string): boolean => {
   const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
   return regex.test(arg)
 }
+
+export const getEmailDomain = (arg: string): string | undefined => {
+  return isEmail(arg) ? arg.toLowerCase().split('@').pop() : ''
+}
+
+export const isPublicDomain = (email: string): boolean => {
```

Comment:
i think it should be hasPublicDomain instead of ispublicDomain

Reply:
Make more sense